### PR TITLE
[WR-111] add create, list, delete realm roles

### PR DESCRIPTION
**Pull request by pirtleshell**
Adds `createRealmRole`, `listRealmRoles`, & `deleteRealmRole` methods to account client.

* establishes a pattern for client methods to be written in ts
* also converts api to ts file. we ignore typechecks for now with `@ts-nocheck` at the top of the file.
* realmRoles is extracted to file external to api
* misc @types packages are added for external dependencies

Further comments inline. Please disregard reviewing compiled files in `dist`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -42,7 +42,9 @@
   },
   "devDependencies": {
     "@toznysecure/sdk": "^1.0.0-rc.4",
+    "@types/isomorphic-fetch": "^0.0.35",
     "@types/jest": "^26.0.24",
+    "@types/uuid": "^8.3.1",
```
Comment (pirtleshell) [inline package.json:47]: when converting code that makes use of external deps, their type definitions sometimes come from [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped) instead of the package itself.

File: src/__tests__/realmRoles.test.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  const sovereignName = 'YassQueen'
+  await client.createRealm(realmName, sovereignName)
+})
+
+afterAll(async () => {
+  await cleanupRealms(client)
+})
+
+describe('Realm Roles', () => {
+  it('creates, lists, & deletes roles', async () => {
```
Comment (pirtleshell) [inline src/__tests__/realmRoles.test.ts:33]: tests for new functionality lives here.

File: src/account.js [hunk 3]
```diff
@@ -3,7 +3,7 @@
  */
 
 const Client = require('./client')
-const API = require('./api')
+const API = require('./api').default
```
Comment (pirtleshell) [inline src/account.js:6]: a quirk of importing ts into js is that it exports as an object with a `default` property. this is how you can access the default export in a js import.

File: src/api/index.ts [hunk 4]
```diff
@@ -1,11 +1,16 @@
+// @ts-nocheck disable type-checking for now. turn me back on when feeling brave.
```
Comment (pirtleshell) [inline src/api/index.ts:1]: i converted this file to ts in name only so that we can begin using ts in it. when the `@ts-nocheck` tag is at the top of the file, errors & warnings are suppressed. this is us tell ts "we know there's a bunch 'wrong' in this file". it will be a gradual process of cleaning it up as we go.

File: src/__tests__/realm.test.js [hunk 5]
```diff
@@ -3,6 +3,7 @@ const Tozny = require('@toznysecure/sdk/node')
 const { v4: uuidv4 } = require('uuid')
 const BasicIdentity = require('../types/basicIdentity')
 const DetailedIdentity = require('../types/detailedIdentity')
+const { cleanupRealms } = require('./utils')
```
Comment (galxy25) [inline src/__tests__/realm.test.js:6]: nice improvement!

File: dist/api/index.js [hunk 6]
```diff
@@ -8,20 +8,26 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, ge
...
-const fetch = require('isomorphic-fetch');
-const Token = require('./token');
-const utils = require('../utils/index');
-const { validateRequestAsJSON, checkStatus } = require('../utils');
-const { DEFAULT_API_URL } = require('../utils/constants');
-// REMOVEME
-const { thisIsATest } = require('./tsIntegrationTest');
+const isomorphic_fetch_1 = __importDefault(require("isomorphic-fetch"));
+const realmRoles_1 = require("./realmRoles");
+const token_1 = __importDefault(require("./token"));
```
Comment (efabens) [inline dist/api/index.js:21]: Out of curiosity why all of the `_1`

File: src/__tests__/realmRoles.test.ts [hunk 7]
```diff
@@ -0,0 +1,77 @@
+import Account from '../account'
+// @ts-ignore no type defs exist for js-sdk
+import Tozny from '@toznysecure/sdk/node'
+import { v4 as uuidv4 } from 'uuid'
+import { cleanupRealms } from './utils'
+import { Role } from '../types'
+
+jest.setTimeout(10000)
```
Comment (galxy25) [inline src/__tests__/realmRoles.test.ts:8]: why is this needed, can we add a comment on the why to this file, move this wait into a test helper?

File: src/account.js (hunk 3 above)
Comment (galxy25) [reply src/account.js:6]: this is a nice pr comment, I think it would be even nicer in this file for posterity

**Review by galxy25 — APPROVED** [review]
nice typing

**Comment by efabens** [comment]
What are the advantages of the `awaiter` style with `yield` vs just using `async` everywhere?

File: dist/api/index.js (hunk 6 above)
Comment (pirtleshell) [reply dist/api/index.js:21]: ha a great question. compiled variable names are a box of mystery i have not yet had a reason to open. the important thing is that it will maintain the names of exports and that these seem to be entirely just internal implementation details.

**Comment by pirtleshell** [comment]
@efabens you seem to be reviewing the compiled code. please disregard everything in `dist`. it is the js that typescript has compiled

**Comment by efabens** [comment]
lol, sounds good

> @efabens you seem to be reviewing the compiled code. please disregard everything in `dist`. it is the js that typescript has compiled

**Review by efabens — APPROVED** [review]
This all makes much more sense.

File: src/__tests__/realmRoles.test.ts (hunk 7 above)
Comment (pirtleshell) [reply src/__tests__/realmRoles.test.ts:9]: added a comment for explanation. if more tests start requiring more time to run, i'll change the global default in the jest config.
